main/src/library/Core/System/Environment.flix

```diff
@@ -0,0 +1,61 @@
+namespace Core/System/Environment {
+  pub def getEnv(): Map[String, String] & Impure =
+    import java.lang.System:getenv();
+    import java.util.Map.entrySet();
+    import java.util.Set.iterator();
+    import java.util.Iterator.hasNext();
+    import java.util.Iterator.next();
+    import java.util.Map$Entry.getKey();
+    import java.util.Map$Entry.getValue();
+    try {
+      let iter = getenv().entrySet().iterator();
+      let iterDone : Unit ~> Bool = () -> not iter.hasNext();
+      let iterNext: Unit ~> (String, String) = () -> {
+        let entry = iter.next() as ##java.util.Map$Entry;
+        let key = entry.getKey() as String;
+        let value = entry.getValue() as String;
+        (key, value)
+      };
+      Iterator.toMap(Iterator(iterDone, iterNext))
```

Comment:
The iterator API is currently (as far as I'm aware) unstable, so this might have to change soon.

Reply:
Yes, might be better with a private helper function that constructs the map explicitly.

Reply:
Hi @magnus-madsen,
I've added a private helper function `def _envMapToFlix(javaMap: ##java.util.Map): Map[String, String] & Impure`.
But by constructing the map explicitly, do you mean without using `Iterator`?

Reply:
Yes, I mean to build the map one element at a time by calling next and so forth.

Reply:
It's done